PolyEngine/UnitTests/Src/TransformComponentTests.cpp

```diff
@@ -0,0 +1,138 @@
+#include <catch.hpp>
+
+#include <TransformComponent.hpp>
+#include <Quaternion.hpp>
+#include <Angle.hpp>
+
+using namespace Poly;
+
+TEST_CASE("TransformComponent with no parent", "[TransformCompoent]") 
+{
+	TransformComponent tc;
+	Vector translationVector = Vector(1, 2, 3);
+	Vector scaleVector = Vector(4, 5, 6);
+	SECTION("Empty constructor")
+	{
+		for (int i = 0; i < 16; ++i)
+		{
+			REQUIRE(tc.GetLocalTransformationMatrix().Data[i] == (i % 5 == 0 ? 1 : 0));
+			REQUIRE(tc.GetGlobalTransformationMatrix().Data[i] == (i % 5 == 0 ? 1 : 0));
+		}
+	}
+
+	// basic transform setting
+	tc.SetLocalTranslation(translationVector);
+	tc.SetLocalScale(scaleVector);
+
+	SECTION("Local transformation")
+	{
+		Matrix tcl = tc.GetLocalTransformationMatrix();
+		REQUIRE(tcl.Data[0] == scaleVector.X);
+		REQUIRE(tcl.Data[3] == translationVector.X);
+		REQUIRE(tcl.Data[5] == scaleVector.Y);
+		REQUIRE(tcl.Data[7] == translationVector.Y);
+		REQUIRE(tcl.Data[10] == scaleVector.Z);
+		REQUIRE(tcl.Data[11] == translationVector.Z);
+		REQUIRE(tc.GetLocalScale() == scaleVector);
+		REQUIRE(tc.GetLocalTranslation() == translationVector);
+	}
+	
+	SECTION("Global transformation")
+	{
+		Matrix tcg = tc.GetGlobalTransformationMatrix();
+		REQUIRE(tcg.Data[0] == scaleVector.X);
+		REQUIRE(tcg.Data[3] == translationVector.X);
+		REQUIRE(tcg.Data[5] == scaleVector.Y);
+		REQUIRE(tcg.Data[7] == translationVector.Y);
+		REQUIRE(tcg.Data[10] == scaleVector.Z);
+		REQUIRE(tcg.Data[11] == translationVector.Z);
+		REQUIRE(tc.GetGlobalScale() == scaleVector);
+		REQUIRE(tc.GetGlobalTranslation() == translationVector);
+	}
+
+	SECTION("Rotation")
+	{
+		TransformComponent tc2;
+		EulerAngles angles = { Angle::FromDegrees(0), 
+			Angle::FromDegrees(0),
+			Angle::FromDegrees(90) };
+		Quaternion rotation(angles);
+		tc2.SetLocalRotation(rotation);
+		float data[] = {0,-1,0,0,
+			  1,0,0,0,
+			  0,0,1,0,
+			  0,0,0,1};
+		for (int i = 0; i < 16; i++)
+		{
+			REQUIRE(tc2.GetLocalTransformationMatrix().Data[i] - data[i] < 0.000001);
```

Comment:
use function bool Cmpf(float, float), which compares two floats and return true if equall.